app/controllers/user_subscriptions_controller.rb

```diff
@@ -5,85 +5,24 @@ class UserSubscriptionsController < ApplicationController
 
   def subscribed
     params.require(%i[source_type source_id])
-    source_type = params[:source_type]
-    source_id = params[:source_id]
-
-    is_subscribed = UserSubscriptions::SubscriptionCacheChecker.call(current_user, source_type, source_id)
+    is_subscribed = UserSubscriptions::SubscriptionCacheChecker.call(current_user, params)
```

Comment:
Maybe we re-name it? `Subscription` is in the name because that's the cache we're checking - the subscriptions.

Reply:
Yeah, I won't push for it super hard, but I think the fact that it's namespaced makes it clear enough that the cache checker is for the (user) subscription. I just found the class name a bit verbose/repetitive here, so was trying to propose an alternative.

Reply:
Sorry! To clarify what I meant...`UserSubscriptions` namespace is for the `UserSubscription` concept/model/object. `SubscriptionCacheChecker` is to check a user's subscriptions in cache.

If we cached more parts of `UserSubscription`, it would start to look something like this (in total, pseudo-names):
- `UserSubscriptions::SubscriptionCacheChecker` _what the user we are checking is subscribed to_
- `UserSubscriptions::SubscribersCacheChecker` _the users subscribed to the user we're checking_
- `UserSubscriptions::SourceCacheCacheker` _the sources the user we're checking is an author of that have subscriptions on them_

Reply:
@atsmith813 Ah, thanks for clarifying this further. Definitely didn't have the whole context here since only the one class exists right now. (Clearly, I'm finding it hard to re-load all of the moving parts of this feature into my brain cache every time I review a liquid tag PR! 😅)

I think we can leave it as it is, and worry about renaming it if we add more "cache checkers". Might be worth adding what you just explained to me (that _Subscription_ `CacheChecker` is checking  _what the user we are checking is subscribed to_) as a comment in the class or where the class is used.

Reply:
@vaidehijoshi I went in to add the comment and went ahead and renamed it to `IsSubscribedCacheChecker` since it returns a boolean. When I originally wrote it, I planned to return the subscriptions. But we ended up going a different direction with how that works. Since we settled on this, I updated it! What do you think? Does this feel more clear now?

Reply:
I like it this better! And it is much more in line with [this check](https://github.com/thepracticaldev/dev.to/pull/9060/commits/62b4684584706137ab975213c8f39f676d173398#diff-2ee6f8c98b7e8296071384bd2ee06ac9R8) now:
```ruby
is_subscribed = UserSubscriptions::IsSubscribedCacheChecker.call(current_user, params)
```